### PR TITLE
[BEAM-14190] Python sends dataflow schema field for pre-portability pipelines

### PR DESCRIPTION
**Pull request by apilloud**
This matches what we do to send schema metadata in Java

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by apilloud** [comment]
run python 3.7 postcommit

**Comment by apilloud** [comment]
Run Python 3.7 PostCommit

**Comment by apilloud** [comment]
R: @TheNeuralBit 

It looks like python post commits are timing out everywhere, I validated the expected structure is showing up in Dataflow's execution logs. Do you know how to fix codecov/patch?

File: sdks/python/apache_beam/coders/row_coder.py [hunk 1]
```diff
@@ -85,6 +87,13 @@ def as_deterministic_coder(self, step_label, error_message=None):
   def to_type_hint(self):
     return self._type_hint
 
+  def as_cloud_object(self, coders_context=None):
```
Comment (TheNeuralBit) [inline sdks/python/apache_beam/coders/row_coder.py:90]: Aren't cloud objects a pre-portability concept that will be deprecated?

Comment (apilloud) [reply sdks/python/apache_beam/coders/row_coder.py:90]: I agree they are pre-portability, I don't know where we are on them being deprecated long term. My understanding is that for portability their generation is moved from the Beam SDK to the Dataflow Service.

This PR is bringing Python up to feature parity with Java. My understanding is that there is no support for Schema metadata in portability yet, but I'll be looking into that next.

**Review by TheNeuralBit — APPROVED** [review]
Could you add some verification? LGTM otherwise

It would be nice if we had some explicit validation to ensure that we're generating equivalent JSON in both Java and Python, like we have for the proto/data serialization format tested with standard_coders.yaml. I'm not sure that's actually necessary though and I'm not aware of a framework where we could put that. Could you at least add some unit tests thhough?

**Comment by apilloud** [comment]
I added a basic test. I think long term we want to move everything to portability, which contains the schema proto, so I don't think adding significant test coverage is really needed.

**Comment by TheNeuralBit** [comment]
> I added a basic test. I think long term we want to move everything to portability, which contains the schema proto, so I don't think adding significant test coverage is really needed.

SGTM, as long as there's *some* coverage to catch trivial mistakes

**Comment by apilloud** [comment]
Run Python PreCommit

**Comment by apilloud** [comment]
Run Python PreCommit

**Pull request merged**